contracts/staking/ServiceStakingBase.sol

```diff
@@ -435,20 +442,80 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
+        uint256[] memory evictServiceIds,
+        uint256[] memory serviceInactivity,
+        uint256 numEvictServices
+    ) internal
+    {
+        // Get the total number of staked services
+        // All the passed arrays have the length of the number of staked services
+        uint256 totalNumServices = evictServiceIds.length;
+
+        // Get arrays of exact sizes
+        uint256[] memory serviceIds = new uint256[](numEvictServices);
+        address[] memory owners = new address[](numEvictServices);
+        address[] memory multisigs = new address[](numEvictServices);
+        uint256[] memory inactivity = new uint256[](numEvictServices);
+        uint256[] memory serviceIndexes = new uint256[](numEvictServices);
+
+        // Fill in arrays
+        uint256 sCounter;
+        uint256 serviceId;
+        for (uint256 i = 0; i < totalNumServices; ++i) {
+            if (evictServiceIds[i] > 0) {
+                serviceId = evictServiceIds[i];
+                serviceIds[sCounter] = serviceId;
+
+                ServiceInfo storage sInfo = mapServiceInfo[serviceId];
+                owners[sCounter] = sInfo.owner;
+                multisigs[sCounter] = sInfo.multisig;
+                inactivity[sCounter] = serviceInactivity[i];
+                serviceIndexes[sCounter] = i;
+                sCounter++;
+            }
+        }
+
+        // Evict services from the global set of staked services
+        uint256 idx;
+        for (uint256 i = numEvictServices - 1; i > 0; --i) {
+            // Decrease the number of services
+            totalNumServices--;
+            // Get the evicted service index
+            idx = serviceIndexes[i];
+            // Assign last service Id to the index that points to the evicted service Id
+            setServiceIds[idx] = setServiceIds[totalNumServices];
+            // Pop the last element
+            setServiceIds.pop();
+        }
+
+        // Deal with the very first element
+        // Get the evicted service index
+        idx = serviceIndexes[0];
+        // Assign last service Id to the index that points to the evicted service Id
+        setServiceIds[idx] = setServiceIds[totalNumServices - 1];
+        // Pop the last element
+        setServiceIds.pop();
+
+        emit ServicesEvicted(epochCounter, serviceIds, owners, multisigs, inactivity);
+    }
+
     /// @dev Checkpoint to allocate rewards up until a current time.
-    /// @return All staking service Ids.
-    /// @return All staking updated nonces.
-    /// @return Number of reward-eligible staking services during current checkpoint period.
-    /// @return Eligible service Ids.
-    /// @return Eligible service rewards.
-    /// @return success True, if the checkpoint was successful.
+    /// @return All staking service Ids (including evicted ones during within a current epoch).
+    /// @return All staking updated nonces (including evicted ones during within a current epoch).
+    /// @return Set of eligible service Ids.
```

Comment:
reward-eligible

Reply:
why do we no longer return the nonces?

Reply:
We return nonces, we don't return the success flag. No need to return that, just can check for the `serviceIds` size.